ultravox/tools/gradio_demo.py

```diff
@@ -37,116 +38,122 @@ def main():
         conversation_mode=True,
     )
 
-    def add_text(chatbot: gr.Chatbot, text: str) -> gr.Chatbot:
-        return chatbot + [(text, None)]
-
-    def add_audio(chatbot: gr.Chatbot, audio: str) -> gr.Chatbot:
-        return chatbot + [((audio,), None)]
-
-    def process_turn(
-        chatbot: gr.Chatbot,
-        prompt: str,
-        audio: Optional[str] = None,
-        max_new_tokens: int = 200,
-        temperature: float = 0,
-    ):
-        # We want to keep the prompt (mixed audio/text instruction) as is in voice mode, but set it to "" in anticipation of new prompt in text mode.
-        prompt_to_return = prompt
-        if audio:
-            if "<|audio|>" not in prompt:
-                prompt += "<|audio|>"
-            sample = datasets.VoiceSample.from_prompt_and_file(prompt, audio)
-        else:
-            sample = datasets.VoiceSample.from_prompt(prompt)
-            prompt_to_return = ""
-
-        if len(sample.messages) != 1:
-            raise ValueError(
-                f"Expected exactly 1 message in sample but got {len(sample.messages)}"
-            )
 
-        output = inference.infer(
-            sample,
-            max_tokens=max_new_tokens,
-            temperature=temperature,
-        )
+def add_text(chatbot: gr.Chatbot, text: str) -> gr.Chatbot:
+    # We set the prompt to "" in anticipation of the next prompt in text mode.
+    return chatbot + [[text, None]], ""
 
-        chatbot = chatbot + [(None, output.text)]
-        return chatbot, gr.update(value=prompt_to_return)
 
-    def process_text(chatbot, prompt, max_new_tokens, temperature):
-        return process_turn(
-            chatbot, prompt, max_new_tokens=max_new_tokens, temperature=temperature
-        )
+def add_audio(chatbot: gr.Chatbot, audio: str) -> gr.Chatbot:
+    # We want to keep the prompt (mixed audio/text instruction) as is in voice mode.
+    return chatbot + [[(audio,), None]]
 
```

Comment:
We also want to display the prompt used in mixed audio/text instruction, as it provides the full context in the chatbot.

```suggestion
def add_audio(chatbot: gr.Chatbot, audio: str, text: str) -> gr.Chatbot:
    # We want to keep the prompt (mixed audio/text instruction) as is in voice mode.
    if text:
        return chatbot + [[text, None]] + [[(audio,), None]]
    else:
        return chatbot + [[(audio,), None]]
```